core/string/node_path.cpp

```diff
@@ -240,19 +240,26 @@ NodePath NodePath::rel_path_to(const NodePath &p_np) const {
 	common_parent--;
 
 	Vector<StringName> relpath;
+	relpath.resize(src_dirs.size() + dst_dirs.size() + 1);
```

Comment:
If you want to optimize further, you can probably avoid the two resizes by computing the exact size using `common_parent`.

Reply:
Yeah, given how `common_parent` is calculated I'm a bit afraid of edge cases, like `common_parent` greater than the size of `src_dir` or `dst_dirs` (and maybe other ones). So for now it seems like a good compromise to make sure it's no more than two allocations, without changing the logic too much.